ascon/src/lib.rs

```diff
@@ -262,6 +264,16 @@ impl AsRef<[u64]> for State {
     }
 }
 
+#[cfg(feature = "zeroize")]
+impl Drop for State {
```

Comment:
`zeroize` 1.6.0 includes a `derive` feature. That could just be `derive(Zeroize,ZeroizeOnDrop)` conditional on the `zeroize` feature.

Reply:
FWIW I had suggested not using it in another similar PR @aewag opened because the implementation is trivial and writing it by hand avoids some heavyweight dependencies like `syn`

Reply:
ACK, then let's leave it like that. But please add an implementation of the `Zeroize` trait so that users of the crate can put states in structs that derive `Zeroize`.

Reply:
I should probably better document this in `zeroize`, but I suggest either implementing `Zeroize` or a combination of a zeroizing `Drop`+`ZeroizeOnDrop`.

What's nice about implementing zeroizing `Drop` but not `Zeroize` is it prevents a sort of use-after-zeroize vulnerability when the sponge is used in an insecure state.

The main reason to implement `Zeroize` instead is to punt the `Drop` handler up a layer which allows the `zeroize` feature to be enabled with no immediate performance impact.

Reply:
I decided to implement `Drop` & `ZeroizeOnDrop` instead of `Zeroize` because it seemed to me more robust to enforce zeroization at a lower layer.

So any user who includes the `ascon` crate "only" has to enable the `zeroize` feature.

I don't have a hard opinion on this matter, hence I can change it if wanted. Let me know.